libsolidity/ExpressionCompiler.cpp

```diff
@@ -813,9 +816,9 @@ void ExpressionCompiler::appendExternalFunctionCall(FunctionType const& _functio
 	}
 }
 
-unsigned ExpressionCompiler::appendArgumentCopyToMemory(TypePointers const& _types,
-														vector<ASTPointer<Expression const>> const& _arguments,
-														unsigned _memoryOffset)
+unsigned ExpressionCompiler::appendArgumentsCopyToMemory(TypePointers const& _types,
```

Comment:
What about making the types optional and do not convert if they are not present?